Add README section for using reflect/code gen combined (#1074)
square/moshi: README.md

both private and protected properties. If you have configured both, generated adapters will be used
on types that are annotated `@JsonClass(generateAdapter = true)`.

> #### Getting the best developer experience out of both reflection and code gen

Reviewer: ```suggestion
#### Getting the best developer experience using both reflection and code gen
```

> If you want to to use code gen but also want to avoid the build time cost of annotation processing, 

Reviewer: ```suggestion
If you want to use code gen but also want to avoid the build time cost of annotation processing, 
```

Author: ![toutou](https://media1.tenor.com/images/37e0c90b9b209afdcd28554f91748669/tenor.gif?itemid=12859336)

> one pattern you can use is to use reflection for dev builds and code gen for production builds.

Reviewer: The repetition of `use` reads a bit weird.

```suggestion
one approach might be to use reflection for development builds and code gen for production builds.
```

In an Android project, such a setup could look like so:

```gradle
dependencies {
  debugImplementation "com.squareup.moshi:moshi-kotlin:{version}"
  kaptRelease "com.squareup.moshi:moshi-kotlin-codegen:{version}"
}
```

Then in a debug-only source set, contribute the reflective `KotlinJsonAdapter` to your app's Moshi
instance creation.

> Another similar pattern in Android is to use reflect only for IDE builds, and reflect for command

Reviewer: ```suggestion
Another similar pattern in Android is to use reflect only for IDE builds, and code gen for command
```

Reviewer: I would add one or two sentences before this alternate approach explaining why one might want to do this:

> While this can speed up development builds by removing the annotation processing step required for code gen, it can cause inconsistencies between development and release builds, which might be an issue if you use your development builds for internal distribution and testing.

That could be considered out of scope for the readme, but I think you could argue that any of this subjective material could be.

line and CI builds. Gradle configuration for this could look like so:

```gradle
dependencies {

>   if (properties.containsKey('android.injected.invoked.from.ide')) {

Reviewer: Using this approach how would someone contribute the `KotlinJsonAdapter` only for builds like this without using a build variant?  Is the suggestion to contribute it regardless and just let it defer to code gen in CI builds?
